### ISSUE TITLE
Git for linux fails to recognize existing git repositories

### ISSUE THREAD

**Issue by theJenix** [Microsoft/BashOnWindows#175]
I was using Windows 8 and git on Cygwin64 to manage some git repositories before upgrading to Windows 10 and preview 14316.  After upgrading, I installed git for linux using apt-get. I can access those repositories using the bash command line, but if I try to execute a git command I get an error `fatal: Not a git repository (or any of the parent directories): .git`

As a workaround, I can reclone the remote repository using git for linux and it recognizes that repository folder going forward.

**Comment by charlescho** [comment]
it must be something to related to Cygwin. I have repos created using git bash for Windows, which is using MINGW64, and I can run all the git commands in those repos in bash on ubuntu on windows.

**Comment by theJenix** [comment]
You may be right, but I think something is still amiss; I can't even access the files in the .git folder created under Cygwin (I get a permission denied error).  If I compare the file permissions of .git/config to one from a repo cloned under bash, they're definitely different and I can't seem to reasonably open up the permissions of the broken repo to get it to work.  Anyone know what a file's permissions must be for me to be able to see it in bash?

**Comment by benhillis** [comment]
Could you try launching bash as an Administrator?  Files in your NTFS mounts /mnt are beholden to NT permission checks.

**Comment by PhilipDaniels** [comment]
Just to chime in, this was one of the first things I checked and I did not experience any problems using the standard UfW git against any repo, including several created using Cygwin git. Nor did I have any problems when I compiled my own version of git 2.8 to use in UfW. Sounds like permissions to me.

**Comment by theJenix** [comment]
@benhillis Sorry for the delay.  When I launch Bash as an administrator, I still get an error but instead of `fatal: index file open failed: Permission denied` it now says `fatal: index file open failed: Permission denied`.  If I cd into .git and try and `cat` one of the files, here's what I see:
`
root@localhost:~/xxx/.git# ls -la
total 44
drwxrwxrwx 2 root root    0 Apr 16 22:24 .
drwxrwxrwx 2 root root    0 Apr 13 15:40 ..
-rwxrwxrwx 1 root root  868 Apr 13 17:37 COMMIT_EDITMSG
-rwxrwxrwx 1 root root  296 Jan  8 03:50 config
-rwxrwxrwx 1 root root   73 Jan  8 03:49 description
-rwxrwxrwx 1 root root  106 Apr 16 22:24 FETCH_HEAD
-rwxrwxrwx 1 root root   23 Jan  8 03:49 HEAD
drwxrwxrwx 2 root root    0 Jan  8 03:49 hooks
-rwxrwxrwx 1 root root 7753 Apr 16 22:24 index
drwxrwxrwx 2 root root    0 Jan  8 03:49 info
drwxrwxrwx 2 root root    0 Jan  8 03:50 logs
-rwxrwxrwx 1 root root  244 Apr 13 17:37 ms-persist.xml
drwxrwxrwx 2 root root    0 Apr 16 22:24 objects
-rwxrwxrwx 1 root root   41 Apr 16 22:24 ORIG_HEAD
drwxrwxrwx 2 root root    0 Jan  8 03:50 refs
root@localhost:~/xxx/.git# cat config
cat: config: Permission denied
` where xxx is a sym link to /mnt/c/Users/MyUser/Documents/xxx.

It definitely feels like a permissions issue: Bash is reporting 777 permissions for file I tried to `cat` and it still won't go through with it.  Just to restate, this was all done in Bash that I ran as an administrator by typing "Bash" in the "Search the web and Windows" box on my start bar, right-clicking on the Bash on Ubuntu for Windows icon, and selecting Run as Administrator.
